lib/jake/array.ex

```diff
@@ -1,5 +1,100 @@
 defmodule Jake.Array do
-  def gen(_) do
-    StreamData.constant([])
+  @type_list [
+    %{"type" => "integer"},
+    %{"type" => "number"},
+    %{"type" => "boolean"},
+    %{"type" => "string"},
+    %{"type" => "null"},
+    nil
+  ]
+
+  @min_items 0
+
+  @max_items 1000
+
+  def gen_array(map, type), do: arraytype(map, map["enum"], map["items"])
+
+  def arraytype(map, enum, items) when enum != nil, do: Jake.gen_enum(enum, "array")
+
+  def arraytype(map, enum, items) when is_list(items) do
+    list = for n <- items, is_map(n), do: Jake.gen_init(n)
+
+    {min, max} = get_min_max(map)
+
+    case map["additionalItems"] do
+      x when (is_boolean(x) and x) or is_nil(x) ->
+        add_additional_items(list, true, max, min)
+
+      x when is_map(x) ->
+        add_additional_items(list, x, max, min)
+
+      _ ->
+        add_additional_items(list, false, max, min)
+    end
+  end
+
+  def arraytype(map, enum, items) when is_map(items) do
+    {min, max} = get_min_max(map)
+    item = Jake.gen_init(items)
+    decide_min_max(map, item, min, max)
+  end
+
+  def get_min_max(map) do
+    min = Map.get(map, "minItems", @min_items)
+    max = Map.get(map, "maxItems", @max_items)
+    {min, max}
+  end
+
+  def arraytype(map, enum, items) when is_nil(items) and is_nil(enum) do
```

Comment:
Possibly could be combined with the previous `arraytype` definition (Assuming we are handling enum like one of the previous comments). 
``` elixir
def list_gen(map, items) when is_nil(items) or is_map(items) do
  item = if is_nil(items), do: get_on_of(), else: Jake.gen_init(items)
  {min, max} = get_min_max(map)
  decide_min_max(map, item, min, max)
end
```